packages/Avalonia/AvaloniaBuildTasks.targets

```diff
@@ -0,0 +1,39 @@
+<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <_AvaloniaUseExternalMSBuild>$(AvaloniaUseExternalMSBuild)</_AvaloniaUseExternalMSBuild>
+    <_AvaloniaUseExternalMSBuild Condition="'$(_AvaloniaForceInternalMSBuild)' == 'true'">false</_AvaloniaUseExternalMSBuild>
+  </PropertyGroup>
+  
+  <UsingTask TaskName="GenerateAvaloniaResourcesTask"
+             AssemblyFile="$(AvaloniaBuildTasksLocation)"
+             />
+
+
+  <Target Name="AddAvaloniaResources" BeforeTargets="ResolveReferences">
+    <PropertyGroup>
+      <AvaloniaResourcesTemporaryFilePath Condition="'$(AvaloniaResourcesTemporaryFilePath)' == ''">$(IntermediateOutputPath)/Avalonia/resources</AvaloniaResourcesTemporaryFilePath>
+    </PropertyGroup>
+    <ItemGroup>
+      <EmbeddedResource Include="$(AvaloniaResourcesTemporaryFilePath)">
+        <LogicalName>!AvaloniaResources</LogicalName>
+      </EmbeddedResource>
+    </ItemGroup>
+  </Target>
+  
+  <Target Name="GenerateAvaloniaResources" 
+          BeforeTargets="CoreCompile;CoreResGen"
+          Inputs="@(AvaloniaResource);$(MSBuildAllProjects)"
+          Outputs="$(AvaloniaResourcesTemporaryFilePath)"
+          DependsOnTargets="$(BuildAvaloniaResourcesDependsOn);AddAvaloniaResources;ResolveReferences">
+    <GenerateAvaloniaResourcesTask
+      Condition="'$(_AvaloniaUseExternalMSBuild)' != 'true'"
+      Output="$(AvaloniaResourcesTemporaryFilePath)"
+      Root="$(MSBuildProjectDirectory)"
+      Resources="@(AvaloniaResource)"
+      EmbeddedResources="@(EmbeddedResources)"/>
+    <Exec 
+      Condition="'$(_AvaloniaUseExternalMSBuild)' == 'true'"
+      Command="dotnet msbuild /nodereuse:false $(MSBuildProjectFile) /t:GenerateAvaloniaResources /p:_AvaloniaForceInternalMSBuild=true /p:Configuration=$(Configuration)"/>
```

Comment:
Unless we go .NET Core only, we shouldn't depend on the presence of the .NET Core SDK in our build scripts in a hidden way like this.

Reply:
One way to fix this would be to use a combination of the `$(MSBuildRuntimeType)` and `$(MSBuildBinPath)` properties to make it sure that we are invoking the same MSBuild exe that the outer build is running on.

Reply:
It's only used in Avalonia solution itself and doesn't affect the user of the package in any shape or form, so I see no problem here. There might be issues with using it with our iOS/Android sample projects though.